core/src/main/java/org/apache/iceberg/PositionBasedDeleteRecord.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import com.google.common.base.MoreObjects;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.avro.specific.SpecificData;
+import org.apache.iceberg.types.Types;
+
+import static org.apache.iceberg.types.Types.NestedField.optional;
+import static org.apache.iceberg.types.Types.NestedField.required;
+
+public class PositionBasedDeleteRecord implements IndexedRecord, SpecificData.SchemaConstructable {
```

Comment:
This looks mostly good to me, but I don't think that it needs to be public. For the others, we use a public interface and a package-private implementation. That way, we can provide readers and writers, but we don't have to maintain all of the interfaces that we use, like `IndexedRecord` here. If we want to remove the Avro reader and use purely Iceberg to construct the records, we can make that change if we use an interface.

Reply:
>  If we want to remove the Avro reader and use purely Iceberg to construct the records, we can make that change if we use an interface.

I don't quite follow this. Could you please help to elaborate?

Reply:
Do we need a separate interface for a 'delete' record or `org.apache.iceberg.data.Record` good enough?

Reply:
I think `org.apache.iceberg.data.Record` is enough. Trying to understanding the context, so since we will update internal Avro reader to use `org.apache.iceberg.data.Record` eventually, both of you are suggesting implement `org.apache.iceberg.data.Record` instead of `IndexedRecord`.  Right?